src/download.test.ts

```diff
@@ -261,6 +261,17 @@ describe("getFileContent", () => {
 
     expect(request.headers["Range"]).toEqual(range);
   });
+
+  it("should register onDownloadProgress callback if defined", async () => {
+    mock.onGet(expectedUrl).replyOnce(200, skynetFileContents, fullHeaders);
+
+    await client.getFileContent(skylink, { onDownloadProgress: jest.fn() });
+
+    expect(mock.history.get.length).toBe(1);
+    const request = mock.history.get[0];
+
+    expect(request.onDownloadProgress).toEqual(expect.any(Function));
```

Comment:
can we assert the exact function stub here?

Reply:
You can't really test for function equality in Javascript. You can do something like [this](https://stackoverflow.com/questions/12216540/how-to-test-for-equality-of-functions-in-javascript/21680065) but it feels hacky to me.

Reply:
the way to do this would be to use `toHaveBeenCalled` on the callback but you would need to execute the download function

Reply:
I'm not sure how to execute it if axios is being mocked. But the way I have it is good enough. If `onDownloadProgress` isn't passed in then `request.onDownloadProgress` will be undefined.

Reply:
I think a lot of the time our tests don't prove anything.  
I know that it works, but you'd have to do the following to really prove that:

```
mock.onGet(expectedUrl).reply(200, skynetFileContents, fullHeaders);

// assert `onDownloadProgress` is not defined if not set
await client.getFileContent(skylink);
expect(mock.history.get.length).toBe(1);
const request = mock.history.get[0];
expect(request.onDownloadProgress).not.toBeDefined();

// assert `onDownloadProgress` is defined when passed as an option
await client.getFileContent(skylink, { onDownloadProgress: jest.fn() });
expect(mock.history.get.length).toBe(2);
const request = mock.history.get[1];
expect(request.onDownloadProgress).toBeDefined();
```

Reply:
That's a good point. I added that check and cleaned up the callback tests a little.